lib/network_monitor.c

```diff
@@ -1021,6 +1028,7 @@ void mptcpd_nm_destroy(struct mptcpd_nm *nm)
                 l_error("Failed to unregister IPv6 monitor.");
 
         l_queue_destroy(nm->interfaces, mptcpd_interface_destroy);
+        nm->interfaces = NULL;
 
         l_netlink_destroy(nm->rtnl);
         l_free(nm);
```

Comment:
Maybe defer `l_free(nm)` until the netlink call destructor runs if needed? If that netlink case is the only one, maybe some flag in the `nm` struct would be enough rather than full-blown refcounting.

Reply:
Perhaps I'm missing something but I don't see why that would be necessary.  At this point in the code we should already know that nothing else is using `nm`.

The one exception I can think of is plugins that run asynchronous operations could be holding on to a reference to `nm` during network monitor destruction.  To address that case, I believe we'd have to defer network monitor destruction until after plugins are unloaded.

Reply:
That case (async operations during destruction) is what I was referring to - I assumed that the two functions modified in the patch were coupled in order to address that scenario. That is, the `l_queue_isempty()` at line 900 wasn't doing the right thing unless `nm->interfaces` was nulled out at line 1031.

Reply:
If this async case doesn't actually happen (or can't happen), then the commit looks fine.

Reply:
I'll adjust the network monitor life cycle so that it always exists for the life of all plugins to at least address the one async exception I could think of.  Hopefully there are no other async issues to be concerned about.

If I get some time I'll take a look at ELL's `l_netlink` command processing code to see if I can make it match the better `l_genl` behavior, and submit a patch upstream.

Reply:
PR #55 contains the changes that extend path manager and network monitor lifetimes beyond plugin lifetimes.